generate-assets/src/lib.rs

```diff
@@ -10,6 +10,53 @@ pub mod gitlab_client;
 
 type CratesIoDb = cratesio_dbdump_csvtab::rusqlite::Connection;
 
+const OFFICIAL_BEVY_CRATES: &'static [&'static str] = &[
```

Comment:
Hmm, I wonder if there's a way to verify that this is kept up to date 🤔

Reply:
@mockersf mentionned using https://github.com/bevyengine/bevy/blob/main/tools/publish.sh here: https://github.com/bevyengine/bevy-website/issues/475#issuecomment-1308738930

I didn't use it because I didn't want to spend too long on this, but I agree that it creates a "we need to permanently update a list somewhere in the code" task which isn't great.

Also it simplifies the code that it's a constant, otherwise you would have to fetch it at a higher level and pass and clone the iterator, I guess?

Also btw now that you point that out I forgot: in the list of the link I see that there is no bevy_a11y for example. I simply did a ls of the `crates` folder of the bevy repo, which does contain bevy_a11y, and this crate exists on crates.io: https://crates.io/crates/bevy_a11y, so I'm not sure which list is correct here?

Reply:
Another potential source of Bevy crates is https://crates.io/users/cart, this info can probably be retrieved from crates.io dump.

But it seems good enough as a first step to use a static list

Reply:
> Another potential source of Bevy crates is https://crates.io/users/cart, this info can probably be retrieved from crates.io dump.

Just implemented this in a new commit.
I tested it nothing changed in the diff.

I copied code from https://github.com/alyti/cratesio-dbdump-lookup/blob/main/src/lib.rs#L341 (which is private and I modified it anyway, the crate haven't been updated since 2021, so? Should I make a PR there to put it there instead?)

I didn't use https://crates.io/users/cart I used homepage = "https://bevyengine.org" and repository = "https://github.com/bevyengine/bevy", it should be enough.

The list it pulls btw:
```
    "bevy",
    "bevy_a11y",
    "bevy_animation",
    "bevy_app",
    "bevy_asset",
    "bevy_audio",
    "bevy_core",
    "bevy_core_pipeline",
    "bevy_derive",
    "bevy_diagnostic",
    "bevy_dylib",
    "bevy_dynamic_plugin",
    "bevy_ecs",
    "bevy_encase_derive",
    "bevy_gilrs",
    "bevy_gltf",
    "bevy_hierarchy",
    "bevy_input",
    "bevy_internal",
    "bevy_log",
    "bevy_macro_utils",
    "bevy_math",
    "bevy_mikktspace",
    "bevy_pbr",
    "bevy_property",
    "bevy_property_derive",
    "bevy_ptr",
    "bevy_reflect",
    "bevy_reflect_derive",
    "bevy_render",
    "bevy_render_macros",
    "bevy_scene",
    "bevy_sprite",
    "bevy_tasks",
    "bevy_text",
    "bevy_time",
    "bevy_transform",
    "bevy_type_registry",
    "bevy_ui",
    "bevy_utils",
    "bevy_wgpu",
    "bevy_window",
    "bevy_winit",
```

I added two new tests to test what happen if there's a problem retrieving the list

Reply:
You could potentially also check for the list at https://github.com/bevyengine/bevy-crate-reservations/blob/main/reserved_crates which should all be reserved by bevy already

Reply:
@IceSentry 

I think the current list is better since:
- I already have the crates.io db dump anyway
- it contains only crates that are or were actually used, reserved_crates have a repository of `https://github.com/bevyengine/bevy-crate-reservations` instead
- nobody should depend on reserved crates, and even if they do, I shouldn't pick a version number from it, I would get version 0.0.1 since it's the version of reserved crates, which is wrong

The only problem with the current list is if some bevy crates are moved to another repo at some point, but even then we could just modify the SQLite query to reflect that.

Reply:
Other (very specific) problem I see: a project depending on an official subcrate on its main branch, but that subcrate is a new yet-unrealised one. The "main" version wouldn't be picked then.